numpy/linalg/linalg.py

```diff
@@ -1615,17 +1615,20 @@ def svd(a, full_matrices=True, compute_uv=True, hermitian=False):
         # reversing is cheap by design in numpy, so we do so to be consistent
         if compute_uv:
             s, u = eigh(a)
-            s = s[..., ::-1]
-            u = u[..., ::-1]
-            # singular values are unsigned, move the sign into v
-            vt = transpose(u * sign(s)[..., None, :]).conjugate()
+            sgn = sign(s)
             s = abs(s)
+            sidx = argsort(s)[..., ::-1]
+            sgn = sgn[..., sidx]
+            s = s[..., sidx]
+            u = u[..., sidx]
```

Comment:
I might be wrong, but do these need to use `take_along_axis`?

Reply:
Axis is always `-1` in the sort, so I really don't think it matters.

Reply:
```python
>>> a = np.arange(12).reshape(3, 4)
>>> a[..., a.argsort()].shape
(3, 3, 4)
>>> np.take_along_axis(a, a.argsort(), axis=-1).shape
(3, 4)
```

It does matter.

Reply:
Looks like `np.lib.index_base` is half-initialized when `linalg` is initialized, so I have to do the import inside the function.